File: common/src/main/java/com/thd/store/dto/BaseResponse.java
package com.thd.store.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@Builder
public class BaseResponse implements Serializable {
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy hh:mm:ss")
    private LocalDateTime timestamp = LocalDateTime.now();
    private Object body;
    private String message;
    private String status;
    private Integer code;
}

File: store-admin/src/app/utils/BaseResponse.ts
export interface BaseResponse{
  timestamp?: Date;
  body?: any;
  message?: string;
  status?: string;
  code?: number;
}
export interface BaseObject {
  name?:string;
  code?:string;
  description?:string;
  shortDescription?:string;
  id?:number;
  voided?:boolean;
}
